### PR TITLE
Summary: Pico2 demo of simple neural network (MNIST)

### PR DESCRIPTION
**Pull request by psiddh**
Test Plan:
1. Setup Conda
2. examples/arm/setup.sh --i-agree-to-the-contained-eula; source examples/ arm/ethos-u-scratch/setup_path.sh
3. cd examples/raspberry_pi/pico2
4. python export_mlp_mnist.py
5.  ./build_firmware_pico.sh --model=balanced_tiny_mlp_mnist.pte
Results:

PREDICTED: 4 (Expected: 4) ✅ CORRECT!

==================================================

=== Digit 7 ===

> 
>               #############
>               #############
>                               ####
>                             #### 
>                           ####  
>                          ####   
>                       ####    
>                    ####  
>   

                                          

Input stats: 159 white pixels out of 784 total
Running neural network inference...
✅ Neural network results:
  Digit 0: 370.000
  Digit 1: 0.000
  Digit 2: -3.000
  Digit 3: -3.000
  Digit 4: 860.000
  Digit 5: -3.000
  Digit 6: -3.000
  Digit 7: 1640.000 ← PREDICTED
  Digit 8: -3.000
  Digit 9: -3.000

� PREDICTED: 7 (Expected: 7) ✅ CORRECT!

==================================================

� All tests complete! PyTorch neural network works on Pico2!

Reviewers:

Subscribers:

Tasks:

Tags:

### Summary
[PLEASE REMOVE] See [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests) for ExecuTorch PR guidelines.

[PLEASE REMOVE] If this PR closes an issue, please add a `Fixes #<issue-id>` line.

[PLEASE REMOVE] If this PR introduces a fix or feature that should be the upcoming release notes, please add a "Release notes: <area>" label. For a list of available release notes labels, check out [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests).

### Test plan
[PLEASE REMOVE] How did you test this PR? Please write down any manual commands you used and note down tests that you have written if applicable.

### CODE REVIEW

**Review by digantdesai — APPROVED** [review]
❤️
